doc/encoding.rdoc

```diff
@@ -132,7 +132,232 @@ returns the \Encoding of the concatenated string, or +nil+ if incompatible:
   s1 = "\xa1\xa1".force_encoding('euc-jp')     # => "\x{A1A1}"
   Encoding.compatible?(s0, s1)                 # => nil
 
-==== \Encoding Options
+=== \String \Encoding
+
+A Ruby String object has an encoding that is an instance of class \Encoding.
+The encoding may be retrieved by method String#encoding.
+
+The default encoding for a string literal is the script encoding
+(see Encoding@Script+encoding):
+
+  's'.encoding # => #<Encoding:UTF-8>
+
+The default encoding for a string created with method String.new is ASCII-8BIT
+if no argument is given, UTF-8 otherwise:
+
+  s = String.new        # => ""
+  s.encoding            # => #<Encoding:ASCII-8BIT>
+  s = String.new('foo') # => "foo"
+  s.encoding            # => #<Encoding:UTF-8>
+
+In either case, any encoding may be specified:
+
+  s = String.new(encoding: 'UTF-8')             # => ""
+  s.encoding                                    # => #<Encoding:UTF-8>
+  s = String.new('foo', encoding: 'ASCII-8BIT') # => "foo"
+  s.encoding                                    # => #<Encoding:ASCII-8BIT>
+
+The encoding for a string may be changed:
+
+  s = "R\xC3\xA9sum\xC3\xA9"     # => "Résumé"
+  s.encoding                     # => #<Encoding:UTF-8>
+  s.force_encoding('ISO-8859-1') # => "R\xC3\xA9sum\xC3\xA9"
+  s.encoding                     # => #<Encoding:ISO-8859-1>
+
+Changing the assigned encoding does not alter the content of the string;
+it changes only the way the content is to be interpreted:
+
+  s                         # => "R\xC3\xA9sum\xC3\xA9"
+  s.force_encoding('UTF-8') # => "Résumé"
+
+The actual content of a string may also be altered;
+see {Transcoding a String}[#label-Transcoding+a+String].
+
+Here are a couple of useful query methods:
+
+  s = "abc".force_encoding("UTF-8")         # => "abc"
+  s.ascii_only?                             # => true
+  s = "abc\u{6666}".force_encoding("UTF-8") # => "abc晦"
+  s.ascii_only?                             # => false
+
+  s = "\xc2\xa1".force_encoding("UTF-8") # => "¡"
+  s.valid_encoding?                      # => true
+  s = "\xc2".force_encoding("UTF-8")     # => "\xC2"
+  s.valid_encoding?                      # => false
+
+=== \Symbol and \Regexp Encodings
+
+The string stored in a Symbol or Regexp object also has an encoding.
+which is returned, respectively, by methods Symbol#encoding or Regexp#encoding;
+the default encoding for these, however, is US-ASCII, not UTF-8:
```

Comment:
I don't think this is always true:

```ruby
:"\u1234".encoding
# => #<Encoding:UTF-8>
/\u1234/.encoding
# => #<Encoding:UTF-8>
```

I think the default encoding is US-ASCII if all characters are US-ASCII.  Otherwise, it depends on the script encoding:

```ruby
(eval "# encoding: iso-8859-1\n/\x87/").encoding
# => #<Encoding:ISO-8859-1>
(eval "# encoding: iso-8859-1\n:\200").encoding
# => #<Encoding:ISO-8859-1>
```

Reply:
Done.